webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteReducer.js

```diff
@@ -0,0 +1,38 @@
+import Immutable from 'seamless-immutable';
+import { createSelector } from 'reselect';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETION_REQUEST,
+  AUTO_COMPLETION_SUCCESS,
+  AUTO_COMPLETION_FAILURE,
+} from './AutoCompleteConstants';
+
+const initialState = Immutable({
+  options: [],
+  status: STATUS.RESOLVED,
+  error: null,
+  searchQuery: '',
+});
+
+export default (state = initialState, action) => {
+  const { payload, type } = action;
+  const optionsSelector = createSelector(
```

Comment:
selectors should be on different files (i.e `SelecorAutoComplete.js`)
and should be used through `mapStateToProps `

You can follow this PR #5778